src/tools/aml-language-server/lspLoader.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import fs from 'fs';
+
+import {AmlServiceContext} from './util.js';
+
+export class LspLoader {
+  context: AmlServiceContext;
+  constructor(context: AmlServiceContext) {
+    this.context = context;
+  }
+
+  path(fileName: string): string {
+    return fileName.replace(/[/][^/]+$/, '/');
+  }
+
+  join(prefix: string, path: string): string {
+    if (/^https?:\/\//.test(path)) {
+      return path;
+    }
+    // TODO: replace this with something that isn't hacky
+    if (path[0] === '/' || path[1] === ':') {
+      return path;
+    }
+    prefix = this.path(prefix);
+    path = this.normalizeDots(`${prefix}${path}`);
+    return path;
+  }
+
+  // convert `././foo/bar/../baz` to `./foo/baz`
```

Comment:
This runs in Node, right? Can't you use https://nodejs.org/api/path.html#path_path_normalize_path ?

Reply:
I've switch this to use node's path lib.